src/fsharp/service/IncrementalBuild.fs

```diff
@@ -1506,188 +583,298 @@ type IncrementalBuilder(tcGlobals, frameworkTcImports, nonFrameworkAssemblyInput
       }
 
     // END OF BUILD TASK FUNCTIONS
-    // ---------------------------------------------------------------------------------------------            
-
-    // ---------------------------------------------------------------------------------------------            
-    // START OF BUILD DESCRIPTION
-
-    // Inputs
-    let fileNamesNode               = InputVector<range*string*(bool*bool)> "FileNames"
-    let referencedAssembliesNode    = InputVector<Choice<string, IProjectReference>*(TimeStampCache -> CompilationThreadToken -> DateTime)> "ReferencedAssemblies"
-        
-    // Build
-    let stampedFileNamesNode        = Vector.Stamp "SourceFileTimeStamps" StampFileNameTask fileNamesNode
-    let stampedReferencedAssembliesNode = Vector.Stamp "StampReferencedAssembly" StampReferencedAssemblyTask referencedAssembliesNode
-    let initialTcAccNode            = Vector.Demultiplex "CombineImportedAssemblies" CombineImportedAssembliesTask stampedReferencedAssembliesNode
-    let tcStatesNode                = Vector.ScanLeft "TypeCheckingStates" (fun ctok tcAcc n -> TypeCheckTask ctok tcAcc (ParseTask ctok n)) initialTcAccNode stampedFileNamesNode
-    let finalizedTypeCheckNode      = Vector.Demultiplex "FinalizeTypeCheck" FinalizeTypeCheckTask tcStatesNode
-
-    // Outputs
-    let buildDescription            = new BuildDescriptionScope ()
+    // ---------------------------------------------------------------------------------------------
 
-    do buildDescription.DeclareVectorOutput stampedFileNamesNode
-    do buildDescription.DeclareVectorOutput stampedReferencedAssembliesNode
-    do buildDescription.DeclareVectorOutput tcStatesNode
-    do buildDescription.DeclareScalarOutput initialTcAccNode
-    do buildDescription.DeclareScalarOutput finalizedTypeCheckNode
-
-    // END OF BUILD DESCRIPTION
-    // ---------------------------------------------------------------------------------------------            
+    let mutable initialTcAccCache = None
```

Comment:
Is this value accessible outside the scope it's declared in?

Reply:
Yes, it's only accessible to the internals of the class though. I wish it could just be a lazy evaluation but it requires an argument, so we have to resort to stuff like this. Maybe there is a memoize function I could use somewhere?